lib/mixins/host-attributes-mixin.html

```diff
@@ -0,0 +1,65 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="property-accessors.html">
+<link rel="import" href="../utils/mixin.html">
+<script>
+  (function() {
+    'use strict';
+
+    /**
+     * Element class mixin to provide default attributes and values, similar to
+     * the "hostAttributes" configuration property used in Polymer 1.x.
+     *
+     * By returning a map from a static class property named "hostAttributes",
+     * the custom element will ensure that the given attributes are set to given value if
+     * that attribute has not previously been set.
+     *
+     * For example, if an element implements hostAttributes like so, an instance of that element
+     * will have an attribute named "foo" with value "bar":
+     *
+     * ```
+     *   class Foo extends Polymer.HostAccessors(Polymer.Element) {
+     *     static get hostAttributes() { return {'foo': 'bar' }}
+     *   }
+     * ```
+     *
+     * @polymerMixin
+     * @mixes Polymer.PropertyAccessors
+     * @memberof Polymer
+     * @summary Element class mixin to provide default attributes and values
+     */
+    Polymer.HostAttributes = Polymer.dedupingMixin((base) => {
+      /**
+       * @constructor
+       * @extends {base}
+       * @implements {Polymer_PropertyAccessors}
+       */
+      let attributeBase = Polymer.PropertyAccessors(base);
+
+      /**
+       * @polymerMixinClass
+       * @implements {Polymer_HostAttributes}
+       */
+      class HostAttributes extends attributeBase {
+        ready() {
+          let hostAttributes = this.constructor.hostAttributes;
```

Comment:
This duplicates code in [legacy-element-mixin](https://github.com/Polymer/polymer/blob/master/lib/legacy/legacy-element-mixin.html#L149). Ideally, it wouldn't and legacy-element-mixin would use this mixin. Alternatively, this could just be moved to another repo and then the implementation can diverge slightly.